src/lib/format/ftl/ast/ast.js

```diff
@@ -0,0 +1,136 @@
+class Node {
+  constructor() {
+    this.type = this.constructor.name;
+  }
+}
+
+class Resource extends Node {
+  constructor() {
+    super();
+    this.body = [];
+  }
+}
+
+class Entry extends Node {
+  constructor() {
+    super();
+  }
+}
+
+class Identifier extends Node {
+  constructor(name) {
+    super();
+    this.name = name;
+  }
+}
+
+class Value extends Node {
+  constructor() {
+    super();
+  }
+}
+
+class String extends Value {
+  constructor(source, content) {
+    super();
+    this.source = source;
+    this.content = content;
+  }
+}
+
+class Trait extends Node {
+  constructor(id, value) {
+    super();
+    this.id = id;
+    this.value = value;
+  }
+}
+
+class Variant extends Node {
```

Comment:
Variants and Traits should be the same:  Members.  `entry.traits` and `placeable.variants` should both be Arrays of Members.